### PR TITLE
CSP-928 AAN Apprentice Directory Message

### PR DESCRIPTION
**Pull request by ManishAgarwalDfE**

### CODE REVIEW

File: src/EmployerAan/SFA.DAS.EmployerAan.Api.UnitTests/Controllers/NotificationsControllerTests/NotificationsControllerTests.cs
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan.Api.UnitTests/Controllers/NotificationsControllerTests/NotificationsControllerTests.cs:1]: I see you have moved the test file into a folder, but usually we do this when we have more than one test file for one SUT. So I guess, you probably had to rename the existing file to GET tests and then add a POST tests in a separate file.

File: src/EmployerAan/SFA.DAS.EmployerAan/InnerApi/Notifications/Requests/PostNotificationRequest.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+namespace SFA.DAS.EmployerAan.Application.InnerApi.Notifications;
+
+public record PostNotificationRequest() : IRequest<GetNotificationResponse>
+{
+    [FromHeader(Name = Constants.ApiHeaders.RequestedByMemberIdHeader)]
+    public Guid RequestedByMemberId { get; set; }
+    [FromQuery]
+    public Guid MemberId { get; set; }
+    [FromQuery]
+    public int NotificationTemplateId { get; set; }
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan/InnerApi/Notifications/Requests/PostNotificationRequest.cs:15]: This attribute decoration is wrong. Remove all the decorations. Also remove the RequestedByMemberId  property as that is injected explicitly

File: src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/NotificationsController.cs [hunk 2]
```diff
@@ -29,4 +31,18 @@ public async Task<IActionResult> GetNotification([FromRoute] Guid id, [FromHeade
         };
         return Ok(result);
     }
+
+    [HttpPost]
+    [ProducesResponseType(typeof(GetNotificationResponse), StatusCodes.Status200OK)]
+    public async Task<IActionResult> CreateNotification([FromHeader(Name = Constants.ApiHeaders.RequestedByMemberIdHeader)] Guid requestedByMemberId, [FromBody] CreateNotificationModel model, CancellationToken cancellationToken)
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/NotificationsController.cs:37]: Model can be dropped and the request body can be forwarded straight to the outer api client

**Review by DevenKShah — CHANGES_REQUESTED** [review]

File: src/EmployerAan/SFA.DAS.EmployerAan.Api.UnitTests/Controllers/NotificationsControllerTests/NotificationsControllerTests.cs
Comment (ManishAgarwalDfE) [reply src/EmployerAan/SFA.DAS.EmployerAan.Api.UnitTests/Controllers/NotificationsControllerTests/NotificationsControllerTests.cs:1]: Done

File: src/EmployerAan/SFA.DAS.EmployerAan/InnerApi/Notifications/Requests/PostNotificationRequest.cs (hunk 1 above)
Comment (ManishAgarwalDfE) [reply src/EmployerAan/SFA.DAS.EmployerAan/InnerApi/Notifications/Requests/PostNotificationRequest.cs:15]: Done

File: src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/NotificationsController.cs (hunk 2 above)
Comment (ManishAgarwalDfE) [reply src/EmployerAan/SFA.DAS.EmployerAan.Api/Controllers/NotificationsController.cs:37]: Done

**Review by MarkFCain — DISMISSED** [review]

File: src/EmployerAan/SFA.DAS.EmployerAan/InnerApi/Notifications/Requests/PostNotificationRequest.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using MediatR;
+using SFA.DAS.EmployerAan.InnerApi.Notifications.Responses;
+
+namespace SFA.DAS.EmployerAan.Application.InnerApi.Notifications;
+
+public record PostNotificationRequest() : IRequest<GetNotificationResponse>
```
Comment (DevenKShah) [inline src/EmployerAan/SFA.DAS.EmployerAan/InnerApi/Notifications/Requests/PostNotificationRequest.cs:6]: Change this to class

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**